internal/tls/tls.go

```diff
@@ -125,6 +125,50 @@ func NewClientConfig(opts ...Option) (*Config, error) {
 		}
 	}
 
+	if c.cfg == nil {
+		c.cfg = &tls.Config{
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blueviolet.svg?style=flat-square)
How about to move tls.Config here and L57-97 into defaultOpts in option.go?

```go

defaultOpts = []Option{
WithTLSConfig(&tls.Config{
...
})
}

```

Reply:
@rinx @kevindiu 
FYI: https://github.com/vdaas/vald/pull/534/files#diff-21cc58b9a1313b78b528877fa29fc290L26

Reply:
@vankichi @kevindiu 
> To avoid same pointer between want and got in test

i can't understand what's the problem. is it in `internal/config/tls_test.go`? or here?

Reply:
@rinx @vankichi 
the problem is the `defaultOpts` is a variable and only initialize once, so the same instance will be `tls.Config{}` will be referenced. 
We are not sure the processing inside `tls` package so it is better to create different instance for it.

Maybe we can change the `defaultOps` to a function to resolve the problem. What do you think?

Reply:
@rinx 
> i can't understand what's the problem
actually we didn't use tls connection for test cluster, but if we start using tls package each connection to DBs & many components will share same tls configuration due to the same pointer which makes problems

Reply:
@kevindiu @kpango 
makes sense.
i agree to Kevin's opinion. it's better to defer the initialization of instances.

Reply:
@rinx Do you mean you agreed to change the `defaultOps` from var to func?

Reply:
yes.

Reply:
Looks good. Thanks!